### ISSUE TITLE
Improve TypeScript type inference for string values

### ISSUE THREAD

**Issue by iiroj** [Avansai/resolve-accept-language#7]
Hello,

thank you for this very useful package! It would be awesome if the type definitions supported a bit nicer inferring of return values. This way one could use literal string values.

Based on my testing something like this in `resolveAcceptLanguage` would already help:

https://github.com/Avansai/resolve-accept-language/blob/07a8a3292358790009ae6f47b06c0087ed76cb02/src/resolve-accept-language.ts#L113-L117

```ts
const resolveAcceptLanguage = <TLocale extends string>(
  acceptLanguageHeader: string,
  locales: TLocale[],
  defaultLocale: TLocale
): TLocale => {
```

It would allow type inference like:

<img width="705" alt="Screenshot 2023-05-08 at 13 27 07" src="https://user-images.githubusercontent.com/11628889/236801126-c444c5e4-bbb4-4120-bb7c-faedb15cc4cd.png">

**Comment by iiroj** [comment]
Hello,

thanks for the update! It looks like the return type is still `string`, should it be `TLocales[number]` instead?

https://github.com/Avansai/resolve-accept-language/blob/7deeb6614357dfc3463ce11ca1fd0ff0b3d4ac85/src/resolve-accept-language.ts#L117

**Comment by nbouvrette** [comment]
My bad you're right @iiroj - the change is available in version [1.1.56](https://github.com/Avansai/resolve-accept-language/releases/tag/v1.1.56)

Thanks
